daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dsom/GroupDef.scala

```diff
@@ -34,70 +34,137 @@ package edu.illinois.ncsa.daffodil.dsom
+            gref.groupDef
+            gdef.groupRef
+            (gref, gdef)
         }
-        Assert.invariant(ggd.length == 1)
-        ggd(0)
+        val res = list(0)
+        res
       }
       case _ => Assert.invariantFailed("not a group")
     }
   }
-
-  override lazy val namedQName = QName.createGlobal(name, targetNamespace, xml.scope)
-
 }
 
-sealed abstract class GlobalGroupDef(xmlArg: Node, schemaDocumentArg: SchemaDocument, val groupRef: GroupRef, position: Int)
-  extends SchemaComponent(xmlArg, schemaDocumentArg)
-  with GlobalNonElementComponentMixin 
-  with NestingTraversesToReferenceMixin {
-
-  requiredEvaluations(modelGroup)
-  requiredEvaluations(validateChoiceBranchKey)
-
-  final lazy val referringComponent = {
-    val res = Some(groupRef)
-    res
-  }
+/**
+ * Many things can be contained either by a model group or a
+ * group definition (aka global sequence group definition or global choice
+ * group definition).
+ *
+ * These things are GroupLike.
+ */
+trait GroupDefLike
+  extends AnnotatedSchemaComponent {
 
-  // final override protected def enclosingComponentDef = groupRef.enclosingComponent
+  def xmlChildren: Seq[Node]
 
   //
-  // Note: Dealing with XML can be fragile. It's easy to forget some of these children
-  // might be annotations and Text nodes. Even if you trim the text nodes out, there are
-  // places where annotations can be.
-  //
-  final lazy val <group>{ xmlChildren @ _* }</group> = xml
-  //
   // So we have to flatMap, so that we can tolerate annotation objects (like documentation objects).
   // and our ModelGroup factory has to return Nil for annotations and Text nodes.
   //
-  final lazy val Seq(modelGroup: ModelGroup) = xmlChildren.flatMap { ModelGroupFactory(_, this, position) }
+  final lazy val groupMembers: Seq[Term] = LV('groupMembers) {
+    val goodXmlChildren = LV('goodXMLChildren) { xmlChildren.flatMap { removeNonInteresting(_) } }.value
+    val positions = List.range(1, goodXmlChildren.length + 1) // range is exclusive on 2nd arg. So +1.
+    val pairs = goodXmlChildren zip positions
+    pairs.flatMap {
+      case (n, i) =>
+        TermFactory(n, this, i)
+    }
+  }.value
+
+  /**
+   * XML is full of uninteresting text nodes. We just want the element children, not all children.
+   */
+  private def removeNonInteresting(child: Node) = {
+    val childList: List[Node] = child match {
+      case _: Text => Nil
+      case _: Comment => Nil
+      case <annotation>{ _* }</annotation> => Nil
+      case _ => List(child)
+    }
+    childList
+  }
+}
+
+/**
+ * Global Group Defs are not actual carriers of annotations syntactically,
+ * but the model groups they contain carry annotations that must be combined
+ * with those from the referring group ref; hence they must be AnnotatedSchemaComponents
+ *
+ * However logic in that base trait takes this behavor of group definitions
```

Comment:
What is this referring to in AnnotatedSchemaComponent? Be specific... check if the code is still in that class/trait. It may have moved.